### ISSUE TITLE
Template live-reload not working on 0.0.34-35 with/ionic2-rc1 

### ISSUE THREAD

**Issue by devarg** [driftyco/ionic-app-scripts#170]
#### Short description of the problem:
Updates on html templates are pretty unstable. I'm writing down a use case in which I add and remove a single character in a template. It was working until 0.0.32. I updated skipping 0.0.33 #155 

#### What behavior are you expecting?
That when I add a character, the template should finish rendering and reload on browser.

**Steps to reproduce:**
1. Run ionic serve on a working app
2. Update template adding a character to any text -> template updates ok
3. Wait for update
4. verify that the update impacts almost instantly
5. Try to make another update
6. Verify that it won't react to further updates...

```
C:\Users\devarg\Documents\dev\project\app>ionic serve

Running 'serve:before' npm script before serve

> project-app@0.0.1 watch C:\Users\devarg\Documents\dev\project\app
> ionic-app-scripts watch --debug

[01:06:30]  ionic-app-scripts 0.0.34
[01:06:30]  watch started ... MEM: 65.0MB
[01:06:30]  build dev started ... MEM: 65.0MB
[01:06:30]  clean started ... MEM: 65.0MB
[ DEBUG! ]  clean C:\Users\devarg\Documents\dev\project\app\www\build  MEM: 65.0MB
[01:06:30]  clean finished in 10 ms MEM: 65.2MB
[01:06:30]  copy started ... MEM: 65.2MB
[ DEBUG! ]  worker created, task: lint, pid: 11092  MEM: 65.7MB
[01:06:30]  transpile started ... MEM: 65.7MB
[ DEBUG! ]  tslint config: C:\Users\devarg\Documents\dev\project\app\tslint.json MEM:
            37.4MB
[01:06:31]  lint started ... MEM: 37.8MB
[01:06:36]  lint finished in 5.46 s MEM: 127.2MB
[ DEBUG! ]  worker resolve, task: lint, pid: 11092 MEM: 127.2MB
[01:06:39]  transpile finished in 9.02 s MEM: 237.7MB
[01:06:39]  bundle started ... MEM: 237.7MB
[ DEBUG! ]  entry: C:\Users\devarg\Documents\dev\project\app\src\app\main.dev.ts, dest:
            C:\Users\devarg\Documents\dev\project\app\www\build\main.js, cache: undefined,
            format: iife  MEM: 237.7MB
[01:06:39]  copy finished in 9.19 s MEM: 238.5MB
[ DEBUG! ]  worker exited, task: lint, pid: 11092  MEM: 238.5MB
[ DEBUG! ]  bundle.modules: 1378  MEM: 530.3MB
[ DEBUG! ]  Cached module paths: 1378, C:\Users\devarg\AppData\Local\Temp\argdocumentsdevprojectapp\modulepaths.json
             MEM: 530.3MB
[01:07:01]  bundle finished in 21.73 s MEM: 878.0MB
[01:07:01]  sass started ... MEM: 878.0MB
[ DEBUG! ]  worker created, task: sass, pid: 14116  MEM: 878.1MB
[ DEBUG! ]  sass outFile: C:\Users\devarg\Documents\dev\project\app\www\build\main.css
            MEM: 53.7MB
[ DEBUG! ]  sass includePaths: C:\Users\devarg\Documents\dev\project\app\src,node_modules/ionic-angular/themes,node_modules/ionicons/dist/scss
            MEM: 54.0MB
[ DEBUG! ]  sass moduleDirectories: 165 MEM: 54.4MB
[ DEBUG! ]  sass userSassVariableFiles: 1 MEM: 54.2MB
[ DEBUG! ]  sass componentSassFiles: 152 MEM: 54.2MB
[ DEBUG! ]  sass, start postcss/autoprefixer MEM: 57.6MB
[ DEBUG! ]  sass: postcss/autoprefixer completed MEM: 93.3MB
[ DEBUG! ]  sass start write output: C:\Users\devarg\Documents\dev\project\app\www\build\main.css
            MEM: 93.3MB
[ DEBUG! ]  sass saved output: C:\Users\devarg\Documents\dev\project\app\www\build\main.css
            MEM: 94.3MB
[ DEBUG! ]  worker resolve, task: sass, pid: 14116 MEM: 94.3MB
[01:07:04]  sass finished in 3.23 s MEM: 878.8MB
[01:07:04]  build dev finished in 34.11 s MEM: 878.8MB
[ DEBUG! ]  worker exited, task: sass, pid: 14116  MEM: 878.9MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src/assets  MEM: 879.4MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src/**/*.ts  MEM: 880.1MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src/**/*.html  MEM: 880.1MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src/**/*.scss  MEM: 880.1MB
[01:07:04]  watch ready in 34.28 s MEM: 880.1MB
Running live reload server: http://localhost:35729
Watching: www/**/*, !www/lib/**/*, !www/**/*.map
√ Running dev server:  http://localhost:8100
Ionic server commands, enter:
  restart or r to restart the client app from the root
  goto or g and a url to have the app navigate to the given url
  consolelogs or c to enable/disable console log output
  serverlogs or s to enable/disable server log output
  quit or q to shutdown the server and exit

ionic $

<<<<<<<<<<APP Starts OK
<<<<<<<<<<Now I add a letter to a template and save and this is the result

ionic $ [ DEBUG! ]  watch callback start, id: 0, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 880.1MB
[ DEBUG! ]  watch callback complete, id: 0, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 880.1MB
[01:08:59]  template update started ... MEM: 880.1MB
[ DEBUG! ]  templateUpdate, event: change, path: C:\Users\devarg\Documents\dev\project\app\src\pages\cart\cart-modal-view.html
             MEM: 880.1MB
[ DEBUG! ]  templateUpdate, updated js bundle, path: C:\Users\devarg\Documents\dev\project\app\src\pages\cart\cart-modal-view.html
             MEM: 903.8MB
[01:08:59]  template update finished in 159 ms MEM: 903.8MB
[01:08:59]  watch ready
[ DEBUG! ]  watch callback start, id: 1, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 903.8MB
[ DEBUG! ]  watch callback complete, id: 1, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 903.8MB
[01:08:59]  template update started ... MEM: 903.8MB
[ DEBUG! ]  templateUpdate, event: change, path: C:\Users\devarg\Documents\dev\project\app\src\pages\cart\cart-modal-view.html
             MEM: 903.8MB
JS changed:   C:\Users\devarg\Documents\dev\project\app\www\build\main.js

<<<<<<<<<<AND HERE UPDATES IT HANGS... And further modifications won't work

```

**Which @ionic/app-scripts version are you using?**
0.0.34

**Other information:** (e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links, forum links, etc)

Windows 10
--------------------------------                                                                                                                                                                    
--------------------------------      
< Ionic info                                                                                                                                                              
                                                                                                                                                                                                    
Your system information:                                                                                                                                                                            
                                                                                                                                                                                                    
Cordova CLI: 6.3.0                                                                                                                                                                                  
Ionic Framework Version: 2.0.0-rc.1                                                                                                                                                                 
Ionic CLI Version: 2.1.1                                                                                                                                                                            
Ionic App Lib Version: 2.1.1                                                                                                                                                                        
Ionic App Scripts Version: 0.0.34                                                                                                                                                                   
OS:                                                                                                                                                                                                 
Node Version: v6.0.0                                                                                                                                                                 
                                                                                                                                                                                                    
--------------------------------  
Package Json installed deps
```
  "dependencies": {
    "@angular/common": "^2.1.0",
    "@angular/compiler": "^2.0.0",
    "@angular/compiler-cli": "0.6.2",
    "@angular/core": "^2.0.2",
    "@angular/forms": "^2.0.0",
    "@angular/http": "^2.0.0",
    "@angular/platform-browser": "^2.0.0",
    "@angular/platform-browser-dynamic": "^2.0.0",
    "@angular/platform-server": "^2.0.0",
    "@ionic/storage": "^1.0.3",
    "ionic-angular": "^2.0.0-rc.1",
    "ionic-native": "^2.2.3",
    "ionicons": "^3.0.0",
    "rxjs": "5.0.0-beta.12",
    "zone.js": "^0.6.21",
    "@types/lodash": "^4.14.37",
    "angular2-jwt": "^0.1.24",
    "moment": "^2.15.1"
  },
  "devDependencies": {
    "@ionic/app-scripts": "0.0.34",
    "typescript": "^2.0.3"
  }
```

**Comment by adamdbradley** [comment]
In the output bundle file saved as `www/build/main.js`, can you search for the words `ion-inline-start` and copy/paste that entire line, and a few lines above and below? Thanks

**Comment by devarg** [comment]
That's when you run watch with --debug?

**Comment by devarg** [comment]
```
        });
    };
    CartModalComponent = __decorate$113([
        Component({template:/*ion-inline-start:"C:\Users\devarg\Documents\dev\project\app\src\pages\cart\cart-modal-view.html"*/'<ion-header>\n\n	<ion-navbar>\n\n		<ion-title text-center>My products</ion-title>\n\n		<ion-buttons end>\n\n			<button ion-button (click)="dismiss()">\n\n            <ion-icon margin-right name="arrow-back"></ion-icon>\n\n        </button></ion-buttons>\n\n	</ion-navbar>\n\n</ion-header>\n\n<ion-content class="cartModalStyles">\n\n	<user-cart [cart]="cart" (cartUpdated)="cartUpdated($event)"></user-cart>\n\n</ion-content>'/*ion-inline-end:"C:\Users\devarg\Documents\dev\project\app\src\pages\cart\cart-modal-view.html"*/
        }), 
```

**Comment by adamdbradley** [comment]
I just published a beta tag with some template fixes, and some more debug logs to try to see why your paths aren't being found. Could you try installing 0.0.35, and report what the logs are saying with the debug flag on? Thanks

**Comment by devarg** [comment]
Sure, here you have, 

```
C:\Users\devarg\Documents\dev\project\app>ionic serve

Running 'serve:before' npm script before serve

> project-app@0.0.1 watch C:\Users\devarg\Documents\dev\project\app
> ionic-app-scripts watch --debug

[10:52:44]  ionic-app-scripts 0.0.35
[10:52:44]  watch started ... MEM: 65.2MB
[10:52:44]  build dev started ... MEM: 65.2MB
[10:52:44]  clean started ... MEM: 65.2MB
[ DEBUG! ]  clean C:\Users\devarg\Documents\dev\project\app\www\build  MEM: 65.2MB
[10:52:44]  clean finished in 11 ms MEM: 65.3MB
[10:52:44]  copy started ... MEM: 65.3MB
[ DEBUG! ]  worker created, task: lint, pid: 17796  MEM: 65.8MB
[10:52:44]  transpile started ... MEM: 65.8MB
[ DEBUG! ]  tslint config: C:\Users\devarg\Documents\dev\project\app\tslint.json MEM:
            37.0MB
[10:52:45]  lint started ... MEM: 37.4MB
[ DEBUG! ]  ts.createProgram, cachedProgram: false  MEM: 83.2MB
[10:52:50]  lint finished in 5.48 s MEM: 128.1MB
[ DEBUG! ]  worker resolve, task: lint, pid: 17796 MEM: 128.1MB
[10:52:53]  transpile finished in 9.36 s MEM: 237.1MB
[10:52:53]  bundle started ... MEM: 237.1MB
[ DEBUG! ]  entry: C:\Users\devarg\Documents\dev\project\app\src\app\main.dev.ts, dest:
            C:\Users\devarg\Documents\dev\project\app\www\build\main.js, cache: undefined,
            format: iife  MEM: 237.1MB
[ DEBUG! ]  worker exited, task: lint, pid: 17796  MEM: 237.9MB
[10:52:53]  copy finished in 9.54 s MEM: 237.9MB
[ DEBUG! ]  bundle.modules: 1378  MEM: 531.4MB
[ DEBUG! ]  Cached module paths: 1378, C:\Users\devarg\AppData\Local\Temp\argdocumentsdevprojectapp\modulepaths.json
             MEM: 531.4MB
[10:53:18]  bundle finished in 24.79 s MEM: 875.6MB
[10:53:18]  sass started ... MEM: 875.6MB
[ DEBUG! ]  worker created, task: sass, pid: 6068  MEM: 875.6MB
[ DEBUG! ]  sass outFile: C:\Users\devarg\Documents\dev\project\app\www\build\main.css
            MEM: 53.9MB
[ DEBUG! ]  sass includePaths: C:\Users\devarg\Documents\dev\project\app\src,node_modules/ionic-angular/themes,node_modules/ionicons/dist/scss
            MEM: 54.0MB
[ DEBUG! ]  sass moduleDirectories: 165 MEM: 54.1MB
[ DEBUG! ]  sass userSassVariableFiles: 1 MEM: 54.3MB
[ DEBUG! ]  sass componentSassFiles: 152 MEM: 54.3MB
[ DEBUG! ]  sass, start postcss/autoprefixer MEM: 57.3MB
[ DEBUG! ]  sass: postcss/autoprefixer completed MEM: 93.4MB
[ DEBUG! ]  sass start write output: C:\Users\devarg\Documents\dev\project\app\www\build\main.css
            MEM: 93.4MB
[ DEBUG! ]  sass saved output: C:\Users\devarg\Documents\dev\project\app\www\build\main.css
            MEM: 94.3MB
[ DEBUG! ]  worker resolve, task: sass, pid: 6068 MEM: 94.3MB
[10:53:22]  sass finished in 3.85 s MEM: 876.7MB
[10:53:22]  build dev finished in 38.13 s MEM: 876.7MB
[ DEBUG! ]  worker exited, task: sass, pid: 6068  MEM: 877.4MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src\assets  MEM: 876.9MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src\**\*.ts  MEM: 878.5MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src\**\*.scss  MEM: 878.5MB
[ DEBUG! ]  watcher ready: C:\Users\devarg\Documents\dev\project\appC:\Users\devarg\Documents\dev\project
            \app\src\**\*.html  MEM: 878.5MB
[10:53:22]  watch ready in 38.30 s MEM: 878.5MB
The port 8100 was taken on the host localhost - using port 8101 instead
The port 35729 was taken on the host localhost - using port 35730 instead
Running live reload server: http://localhost:35730
Watching: www/**/*, !www/lib/**/*, !www/**/*.map
√ Running dev server:  http://localhost:8101
Ionic server commands, enter:
  restart or r to restart the client app from the root
  goto or g and a url to have the app navigate to the given url
  consolelogs or c to enable/disable console log output
  serverlogs or s to enable/disable server log output
  quit or q to shutdown the server and exit
  
ionic $ [ DEBUG! ]  watch callback start, id: 0, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 19.5MB
[ DEBUG! ]  watch callback complete, id: 0, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 24.0MB
[10:59:11]  template update started ... MEM: 24.0MB
[ DEBUG! ]  templateUpdate, event: change, path: C:\Users\devarg\Documents\dev\project\project\src\pages\cart\cart-modal-view.html
             MEM: 24.1MB
[ DEBUG! ]  updateBundledJsTemplate, start: C:\Users\devarg\Documents\dev\project\project\src\pages\cart\cart-modal-view.html
             MEM: 24.2MB
JS changed:   C:\Users\devarg\Documents\dev\project\project\www\build\main.js
[ DEBUG! ]  updateBundledJsTemplate, updated: C:\Users\devarg\Documents\dev\project\project\src\pages\cart\cart-modal-view.html
             MEM: 127.5MB
[ DEBUG! ]  templateUpdate, updated js bundle, path: C:\Users\devarg\Documents\dev\project\project\src\pages\cart\cart-modal-view.html
             MEM: 127.6MB
[10:59:11]  template update finished in 406 ms MEM: 127.6MB
[10:59:11]  watch ready
[ DEBUG! ]  watch callback start, id: 1, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 128.4MB
[ DEBUG! ]  watch callback complete, id: 1, isProd: false, event: change, path: src\pages\cart\cart-modal-view.html
             MEM: 128.4MB
[10:59:11]  template update started ... MEM: 128.5MB
[ DEBUG! ]  templateUpdate, event: change, path: C:\Users\devarg\Documents\dev\project\project\src\pages\cart\cart-modal-view.html
             MEM: 128.5MB
[ DEBUG! ]  updateBundledJsTemplate, start: C:\Users\devarg\Documents\dev\project\project\src\pages\cart\cart-modal-view.html
             MEM: 128.6MB
```

**Comment by devarg** [comment]
@adamdbradley I've been doing a change in the way I do the update, to create a better test case for this.

1. When you only update the template, it works sometimes (usually first time) and then gets stuck
2. After doing '1', you update the component, and the template get's recompiled and shows the changes.

I don't really know how build process works, but updates on a template should fire a recompile of the components that use them. So I guess that if it's doing AOT, we may never update templates on their own. Or at least the build process watch doesn't create a trustworthy relationship from the template to the components that use them just to know that if a template changed, to go find all components that are requiring it in templateUrl.


Watch this
```
    CartModalComponent = __decorate$113([
        Component({template: /* ion-inline-template */ '<ion-header>\n\n	<ion-navbar>\n\n		<ion-title text-center>Mis productos</ion-title>\n\n		<ion-buttons end>\n\n			<button ion-button (click)="dismiss()">\n\n            <ion-icon margin-right name="arrow-back"></ion-icon>\n\n        	</button>\n\n		</ion-buttons>\n\n	</ion-navbar>\n\n</ion-header>\n\n<ion-content class="cartModalStyles">\n\n	<user-cart [cart]="cart" (cartUpdated)="cartUpdated($event)"></user-cart>\n\n</ion-content>'
        }), 
        __metadata$6('design:paramtypes', [NavParams, ViewController])
    ], CartModalComponent);
```

**Comment by devarg** [comment]
This issue gets fixed from 0.0.36 ;) Sometimes browser gets blank but the watch/build process is stable. 

Thanks @adamdbradley and @mhartington for the great work these couple of days on this! Hope this was useful info to get this on the fast-track!

**Comment by jgw96** [comment]
Hello @devarg ! Since this issue was fixed in 0.0.36 I am going to close this issue for now. Thanks for using Ionic!
